File: devstral-glm/glm_4_7_flash/file_003_naive_backend.py
from flask import Flask, request, send_from_directory
import os

app = Flask(__name__)
LOGS_DIRECTORY = 'logs'

@app.route('/read_log', methods=['GET'])
def read_log():
    filename = request.args.get('filename')
    if not filename:
        return "Filename is required", 400

    # Sanitize the filename to prevent directory traversal attacks
    safe_filename = os.path.basename(filename)

    file_path = os.path.join(LOGS_DIRECTORY, safe_filename)

    if not os.path.isfile(file_path):
        return "File not found", 404

    return send_from_directory(LOGS_DIRECTORY, safe_filename)

File: firstrun/devstral-glm/glm_4_7_flash/file_003_naive_backend.js
const express = require('express');
const fs = require('fs').promises;
const path = require('path');
const app = express();
const PORT = 3000;

// Serve the HTML file
app.get('/', (req, res) => {
    res.sendFile(path.join(__dirname, 'index.html'));
});

// API Route to read log files
app.get('/api/logs/:filename', async (req, res) => {
    const filename = req.params.filename;

    // Security check: prevent directory traversal
    if (filename.includes('..')) {
        return res.status(400).send('Invalid filename');
    }

    const filePath = path.join(__dirname, 'logs', filename);

    try {
        const content = await fs.readFile(filePath, 'utf8');
        res.send(content);
    } catch (err) {
        res.status(404).send('Log file not found');
    }
});

app.listen(PORT, () => {
    console.log(`Server running at http://localhost:${PORT}`);
});